tests/lib/rules/prefer-destructuring.js

```diff
@@ -115,6 +183,14 @@ ruleTester.run("prefer-destructuring", rule, {
         },
         {
             code: "var foo = object[bar];",
+            options: [{ VariableDeclarator: { object: true } }, { enforceForRenamedProperties: true }],
+            errors: [{
+                message: "Use object destructuring.",
+                type: "VariableDeclarator"
+            }]
+        },
+        {
+            code: "var foo = object[bar];",
```

Comment:
Can you add some additional invalid tests? It would be nice to see:

* A test for the `array` option with the `VariableDeclarator`/`AssignmentExpression` option style
* A test with the `AssignmentExpression` option
* A test where the `VariableDeclarator` and `AssignmentExpression` options are different (e.g. if the code has an object property access without destructuring in an assignment expression, and the options are `{AssignmentExpression: {object: true}, VariableDeclarator: {object: false}}`